### PR TITLE
add-sidebar-72

### LINKED ISSUE #72: Spike out adding a sidebar
Try using an iframe to create a sidebar that can utilize the whole height of the browser window to display content.

### PR DESCRIPTION
**Pull request by terrycojones**
ceronman:**approve**

Add simple sidebar with side and width options.

Fixes #72

### CODE REVIEW

File: sidebar.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    else {
+        console.log('Sidebar logic error: showSidebar ' +
+                    'called, but no sidebar element exists.');
+    }
+};
+
+var toggleSidebar = function(){
+    var sidebar = document.getElementById('fi_sidebar');
+    console.log('sidebar toggle. element is');
+    console.log(sidebar);
```
Comment (ceronman) [inline sidebar.js:30]: You can use a single console.log statemente here:

`console.log('sidebar toggle. element is', sidebar)`

File: sidebar.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        chrome.extension.sendRequest(
+            {action: 'get-settings'},
+            function(options){
+                console.log('Creating new sidebar element.');
+                console.log(options);
+                sidebar = document.createElement('iframe');
+                sidebar.id = 'fi_sidebar';
+                sidebar.classList.add('sidebar');
+                sidebar.classList.add(options.sidebarSide);
+		sidebar.setAttribute('width', options.sidebarWidth + 'px');
```
Comment (ceronman) [inline sidebar.js:80]: This is not properly indented.

File: sidebar.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+     */
+    var sidebar = document.getElementById('fi_sidebar');
+    
+    if (sidebar){
+        // The sidebar element already exists.
+        // TODO: perhaps do nothing if the currentAbout is the one we were passed?
+        // Setting the value likely causes a reload.
+        console.log('Found existing sidebar element, updating its URL.');
+        currentAbout = about;
+        sidebar.src = 'http://new.fluidinfo.com/infomaniac/' + encodeURIComponent(about);
```
Comment (ceronman) [inline sidebar.js:57]: It'd be nice to do something in the future to avoid hardcoding the src of the sidebar.

**Comment by ceronman** [comment]
+1.

File: sidebar.js (hunk 1 above)
Comment (terrycojones) [inline sidebar.js:30]: This is fixed in a later branch (console logging is removed).

File: sidebar.js (hunk 2 above)
Comment (terrycojones) [inline sidebar.js:80]: Fixed in later branch.

File: sidebar.js (hunk 3 above)
Comment (terrycojones) [inline sidebar.js:57]: Yes, agreed.  Next branch at least has the URL updating in a function.

**Pull request merged**